cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server.go

```diff
@@ -43,43 +50,95 @@ type Server struct {
 	clientGetter             clientGetter
 	globalPackagingNamespace string
 	globalPackagingCluster   string
+
+	// We keep a restmapper to cache discovery of REST mappings from GVK->GVR.
+	restMapper meta.RESTMapper
+
+	// kindToResource is a function to convert a GVK to GVR with
+	// namespace/cluster scope information. Can be replaced in tests with a
+	// dummy version using the unsafe helpers while the real implementation
+	// queries the k8s API for a REST mapper.
+	kindToResource func(meta.RESTMapper, schema.GroupVersionKind) (schema.GroupVersionResource, meta.RESTScopeName, error)
+}
+
+// createRESTMapper returns a rest mapper configured with the APIs of the
+// local k8s API server. This is used to convert between the GroupVersionKinds
+// of the resource references to the GroupVersionResource used by the API server.
+func createRESTMapper() (meta.RESTMapper, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	// To use the config with RESTClientFor, extra fields are required.
+	// See https://github.com/kubernetes/client-go/issues/657#issuecomment-842960258
+	config.GroupVersion = &schema.GroupVersion{Group: "", Version: "v1"}
+	config.NegotiatedSerializer = serializer.WithoutConversionCodecFactory{CodecFactory: scheme.Codecs}
+	client, err := rest.RESTClientFor(config)
+	if err != nil {
+		return nil, err
+	}
+	discoveryClient := discovery.NewDiscoveryClient(client)
```

Comment:
Here we still create a new discovery, but as Michael did in the past, we are using the RESTMapper just in production. For testing, we mock the entire `kindToResource` function